docs/guidance/tls.md

```diff
@@ -0,0 +1,140 @@
+            -sha256 -days 3650 \
+            -nodes \
+            -newkey rsa:2048 \
+            -subj "/CN=*.kuberay.com/C=US/L=San Francisco" \
+            -keyout ca.key -out ca.crt
+
+# Step 1-2: Use `cat $FILENAME | base64` to encode `ca.key` and `ca.crt`.
+#           Then, paste the encoding strings to the Kubernetes Secret in `ray-cluster.tls.yaml`.
+```
+* `ca.key`: CA's private key
+* `ca.crt`: CA's self-signed certificate 
+
+This step is optional because the `ca.key` and `ca.crt` files have
+already been included in the Kubernetes Secret specified in [ray-cluster.tls.yaml](../../ray-operator/config/samples/ray-cluster.tls.yaml).
+
+# Step 2 (Optional): Create separate private key and self-signed certificate for Ray Pods
+
+In [ray-cluster.tls.yaml](../../ray-operator/config/samples/ray-cluster.tls.yaml), each Ray
+Pod (both head and workers) generates its own private key file (`tls.key`) and self-signed
+certificate file (`tls.crt`) in its init container. We generate separate files for each Pod 
+because worker Pods do not have deterministic DNS names, and we cannot use the same 
+certificate across different Pods.
+
+In the YAML file, you'll find a ConfigMap named `tls` that contains two shell scripts: 
+`gencert_head.sh` and `gencert_worker.sh`. These scripts are used to generate the private key 
+and self-signed certificate files (`tls.key` and `tls.crt`) for the Ray head and worker Pods. 
+
+Please find below a brief explanation of what happens in each of these scripts:
+1. A 2048-bit RSA private key is generated and saved as `/etc/ray/tls/tls.key`.
+2. A Certificate Signing Request (CSR) is generated using the private key file (`tls.key`)
+and the `csr.conf` configuration file.
+3. A self-signed certificate (`tls.crt`) is generated using the private key of the 
+Certificate Authority (`ca.key`) and the previously generated CSR.
+
+The only difference between `gencert_head.sh` and `gencert_worker.sh` is the `[ alt_names ]`
+section in `csr.conf` and `cert.conf`. The worker Pods use the fully qualified domain name 
+(FQDN) of the head Kubernetes Service to establish a connection with the head Pod. 
+Therefore, the `[alt_names]` section for the head Pod needs to include the FQDN of the head 
+Kubernetes Service. By the way, the head Pod uses `$POD_IP` to communicate with worker Pods.
+
+```sh
+# gencert_head.sh
+[alt_names]
+DNS.1 = localhost
+DNS.2 = $FQ_RAY_IP
+IP.1 = 127.0.0.1
+IP.2 = $POD_IP
+
+# gencert_worker.sh
+[alt_names]
+DNS.1 = localhost
+IP.1 = 127.0.0.1
+IP.2 = $POD_IP
+```
+
+In [Kubernetes networking model](https://github.com/kubernetes/design-proposals-archive/blob/main/network/networking.md#pod-to-pod), the IP that a Pod sees itself as is the same IP that others see it as. That's why Ray Pods can self-register for the certificates.
+
+# Step 3 (Optional): Configure environment variables for Ray TLS authentication
+
+To enable TLS authentication in your Ray cluster, set the following environment variables:
```

Comment:
Can we clarify where and when these environment variables need to be set?

Reply:
Remove "optional" from the document, and explain what is covered by `ray-cluster.tls.yaml` instead.

https://github.com/ray-project/kuberay/pull/989/commits/d4b70a8b8e3182c014bc7cd35b33b5a320d785fb